tasklist/els-schema/src/main/java/io/camunda/tasklist/util/IndexSchemaValidatorUtil.java

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.tasklist.util;
+
+import static io.camunda.zeebe.client.impl.Loggers.LOGGER;
+
+import com.google.common.collect.Maps;
+import io.camunda.tasklist.exceptions.TasklistRuntimeException;
+import io.camunda.tasklist.property.TasklistProperties;
+import io.camunda.tasklist.schema.IndexMapping;
+import io.camunda.tasklist.schema.IndexMapping.IndexMappingProperty;
+import io.camunda.tasklist.schema.IndexMappingDifference;
+import io.camunda.tasklist.schema.SemanticVersion;
+import io.camunda.tasklist.schema.indices.IndexDescriptor;
+import io.camunda.tasklist.schema.manager.SchemaManager;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+public abstract class IndexSchemaValidatorUtil {
```

Comment:
Can we make this a Component on it is own, where you auto-wire SchemaManager:
- to favor Composition over Inheritance
- to avoid passing SchemaManager between method
- to avoid the usage of static methods

Reply:
We have separate components, one for ElasticSeach OS and ES  where wea re using the clients, which keep the consistency also with our class structure over OS or ES
We have 2 components already for separated ones, we cannot abstract all methods of IndexSchema Validator. I would prefer keep the classes separated and use the Utils. 

Is there any significant gain for not use static methods?

Reply:
it is mostly a matter of readability and reusability of the methods, which is lowered with these too many arguments. If we use Inversion Of Control principle, it becomes easier to reuse
it is becoming similar to `ElasticsearchUtil` which is sometimes not easy to change or to extend

We can have a single `IndexSchemaValidator` component if we create a common interface for `retryElasticsearchClient ` and `retryOpenSearchClient`

Reply:
I did the class preparation over this commit: https://github.com/camunda/camunda/pull/18896/commits/fc28bbb058286af3b9b9c66945a528178f371f0f

Also the follow up issue: https://github.com/camunda/team-hto/issues/626 as we agreed this refactor is out of scope of this epic.